cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -405,19 +455,49 @@ validateTxAuxScripts era files =
         | ScriptFile file <- files ]
       return (TxAuxScripts AuxScriptsInMaryEra scripts)
 
-validateTxWithdrawals :: CardanoEra era
-                      -> [(StakeAddress, Lovelace)]
-                      -> ExceptT ShelleyTxCmdError IO (TxWithdrawals era)
+validateTxWithdrawals
+  :: forall era. IsCardanoEra era
+  => CardanoEra era
+  -> [(StakeAddress, Lovelace, Maybe ScriptFile)]
+  -> ExceptT ShelleyTxCmdError IO (TxWithdrawals BuildTx era)
 validateTxWithdrawals _ [] = return TxWithdrawalsNone
 validateTxWithdrawals era withdrawals =
-    case withdrawalsSupportedInEra era of
-      Nothing -> txFeatureMismatch era TxFeatureWithdrawals
-      Just supported -> return (TxWithdrawals supported withdrawals)
-
-
-validateTxCertificates :: CardanoEra era
-                       -> [CertificateFile]
-                       -> ExceptT ShelleyTxCmdError IO (TxCertificates era)
+  case withdrawalsSupportedInEra era of
+    Nothing -> txFeatureMismatch era TxFeatureWithdrawals
+    Just supported -> do
+      convWithdrawals <- mapM convert withdrawals
+      return (TxWithdrawals supported convWithdrawals)
+ where
+  convert
+    :: (StakeAddress, Lovelace, Maybe ScriptFile)
+    -> ExceptT ShelleyTxCmdError IO
+           (StakeAddress, Lovelace, BuildTxWith BuildTx (Witness WitCtxStake era))
+  convert (sAddr, ll, mScriptFile) =
+    case mScriptFile of
+      Just (ScriptFile fp) -> do
+        ScriptInAnyLang sLang script <- firstExceptT ShelleyTxCmdReadJsonFileError
+                                          $ readFileScriptInAnyLang fp
+        case scriptLanguageSupportedInEra era sLang of
+          Just sLangInEra ->
+            case script of
+              SimpleScript sVer sScript ->
+                return ( sAddr
+                       , ll
+                       , BuildTxWith
+                           (ScriptWitness ScriptWitnessForStakeAddr
+                             $ SimpleScriptWitness sLangInEra sVer sScript)
+                       )
```

Comment:
Looks like we could probably simplify later with some helpful utility function for making the script witness from the script (and era or something). Adding this util before we add plutus script & witness support will probably keep the size of the code here managable, otherwise we'll be duplicating the same logic in several places.